crates/allocator/src/bump.rs

```diff
@@ -0,0 +1,104 @@
+// Copyright 2018-2021 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! A simple bump allocator.
+//!
+//! It's goal to have a much smaller footprint than the admittedly more full-featured `wee_alloc`
+//! allocator which is currently being used by ink! smart contracts.
+//!
+//! The heap which will be used by this allocator is a single page of memory, which in Wasm is
+//! 64KiB. We do not expect contracts to use more memory than this (for now), so we will throw an
+//! OOM error instead of requesting more memory.
+
+use core::alloc::{
+    GlobalAlloc,
+    Layout,
+};
+
+/// A page in Wasm is 64KiB
+const PAGE_SIZE: usize = 64 * 1024;
+
+lazy_static::lazy_static! {
+    pub static ref BUMP_ALLOC: Locked<BumpAllocator> = Locked::new(BumpAllocator::new());
```

Comment:
This is unused in this PR. I guess some leftover of some experiment?

Reply:
Yeah, it was from some previous attempts